core/src/main/java/com/spotify/metrics/net/NetstatMetricSet.java

```diff
@@ -0,0 +1,130 @@
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.spotify.metrics.core.DerivedLongGauge;
+import com.spotify.metrics.core.MetricId;
+import com.spotify.metrics.core.SemanticMetricSet;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class NetstatMetricSet implements SemanticMetricSet {
+
+    private static final String SPACES = "\\s+";
+    private static final String PROC_SNMP_PATH = "/proc/net/snmp";
+    private static final String PROC_NETSTAT_PATH = "/proc/net/netstat";
+
+    private final Supplier<Data> dataSupplier;
+
+    public NetstatMetricSet() {
+        this(Suppliers.memoizeWithExpiration(NetstatMetricSet::readData, 10, TimeUnit.SECONDS));
+    }
+
+    @VisibleForTesting
+    NetstatMetricSet(final Supplier<Data> dataSupplier) {
+        this.dataSupplier = checkNotNull(dataSupplier, "dataSupplier");
+    }
+
+    @Override
+    public Map<MetricId, Metric> getMetrics() {
+        final Map<MetricId, Metric> gauges = new HashMap<>();
+        gauges.put(
+            MetricId.build().tagged("what", "segments-retransmitted", "unit", "tcp-segments/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().segmentsRetransmitted;
+                }
+            });
+        gauges.put(
+            MetricId.build().tagged("what", "bad-segments-received", "unit", "tcp-segments/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().badSegmentsReceived;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-timeouts", "unit", "timeouts/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpTimeouts;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-lost-retransmit", "unit", "retransmit/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpLostRetransmit;
+                }
+            });
+        gauges.put(MetricId.build().tagged("what", "tcp-syn-retrans", "unit", "retransmit/s"),
+            new DerivedLongGauge() {
+                @Override
+                public Long getNext() {
+                    return dataSupplier.get().tcpSynRetrans;
+                }
+            });
+
+        return gauges;
+    }
+
+    private static Data readData() {
+        return readData(PROC_NETSTAT_PATH, PROC_SNMP_PATH);
+    }
+
+    @VisibleForTesting
+    static Data readData(final String netstatPath, final String snmpPath) {
```

Comment:
nitty-pick: instead of having this as a static method with `@VisibleForTesting` what do you think about moving to it's own class that has the `readData` method? This way you could also have these 2 parameters to be fields in it and avoid using static

Reply:
Could you give an example how that will look?